demo.py

```diff
@@ -62,6 +62,13 @@
   
   # Run all the scenarios
   automator.run(use_vhil = use_vhil)
+
```

Comment:
to summarize: this creates an updating list of scenarios that failed as the simulation is run, and then outputs the failures in demo at the end of the run instead of crashing midway